src/main/java/uk/gov/hmcts/reform/sandl.snlevents/service/ProblemService.java

```diff
@@ -57,7 +57,8 @@
             return transformed;
         };
 
-    public final Function<CreateProblem, Problem> problemCreateToDb = (CreateProblem cp) -> {
+    //todo removed final, can't mock final function, discuss how we should handle that
+    public Function<CreateProblem, Problem> problemCreateToDb = (CreateProblem cp) -> {
```

Comment:
true, on one hand, this method should also be static final and not mockable as it is quite simple.
On the other, it does too much(in its body it's referencing another method) and as such, mocking it would be quite useful, and if we go this path it should be a regular method instead of lambda.
For tests leave it as non-final, and I kindly ask to leave a comment there to transfer it into a method in a separate class (not a //TODO as sonar will moan about it).